docs/content/concepts/partitions-schedules-sensors/partitions.mdx

```diff
@@ -133,6 +134,49 @@ result = materialize(
 )
 ```
 
+#### Dynamically partitioned assets (Experimental)
+
+<Note>
+  This feature is <strong>experimental</strong>. Until this feature is marked as
+  stable, partition created prior may not be retained.
```

Comment:
Is it the case in Cloud as well that partitions created may not be retained?

Reply:
Depends on to what extent we want to support existing users using the feature. In Cloud we could always run a data migration to retain existing data

Reply:
If we're writing a data migration, would it not work in open source as well?

Reply:
I think it would work. We would have to make sure users run the data migration if they are affected, though I've never added a data migration in OSS so not super sure about the specifics

Reply:
We're just not sure if the schema that currently exists will require major changes in the future. Managing backcompat state / data migrations in open source feels tricky given that we can't force everyone to migrate, so I'd like to avoid a situation where we need to do this especially since this is an experimental feature

Reply:
@prha - as someone who has done data migrations in open source, curious if you have thoughts here?

My instinct is that it's fine to say something like "Until this feature is marked stable, you maybe required to run migrations to ensure that it keeps functioning." But wouldn't be surprised if there are hidden complexities even with that?

Reply:
I don't think there are any hard and fast rules.  A lot of it depends on how much data there is and how bad of a product experience it would be to run the migration.  I have not ever pre-warned about migrations though.

Typically, for data migrations over the event log, I'll do a hard cutover to the new data format and ignore the old data format.   So then in the release notes, I'd say something like, changed experimental data format blahblahblah, and then instructions in case they want to run the data migration (using `dagster instance reindex`) so that their old data shows up.   If it's not experimental, I might do something like a backcompat query and lazily migrate on reads.

For data migrations over run storage, I'll add them to the list of data migrations that automatically run when you run `dagster instance migrate` (which typically is just schema migrations).